### PR TITLE
Remove LdLib Compass Support

### PR DESCRIPTION
**Pull request by Ghostipedia**
## What
Removes All Compass Content and Integration in GTCEU as a whole, as part of moving away from LDLIB reliance.

Compass is used nearly as frequently as Fabric was for us, but contributed to a bunch of code that had to still be maintained, slimming this down will mean less reliance on other project libraries. Myself and Several people are in favor of this (Namely Tech, Spicy, Onion, and myself)

## Implementation Details
Fully Erases compass and all related content to it

## Outcome
Removes Compass

## Potential Compatibility Issues
If you are the 0.1% of people using compass, you will no longer be able to use it next major update, please look into using Patchouli or Modonomicon if you need a booklet guide.

## Additional Information
i was at house, eating chicken... when phone ring...
"Compass is kil"
"Yes..."
366323089-506e888e-af82-43a2-a262-bce5e9495400

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/common/data/machines/GCYMMachines.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ public static void init() {}
                     .abilities(PartAbility.PARALLEL_HATCH)
                     .workableTieredHullRenderer(GTCEu.id("block/machines/parallel_hatch_mk" + (tier - 4)))
                     .tooltips(Component.translatable("gtceu.machine.parallel_hatch_mk" + tier + ".tooltip"))
-                    .compassNode("parallel_hatch")
+
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/common/data/machines/GCYMMachines.java]: remove these newlines you left

**Review by screret — DISMISSED** [review]
:waaaaaaahhhhhhh:

**Review by YoungOnionMC — APPROVED** [review]

**Comment by krossgg** [comment]
image

**Comment by Spicierspace153** [comment]
O7

**Comment by JuiceyBeans** [comment]
where were you when compass was kil. o7

**Comment by Deepacat** [comment]
it wasn't that bad :(

**Comment by Ghostipedia** [comment]
> it wasn't that bad :(

https://www.youtube.com/watch?v=Xsrjy7140Lw

**Comment by omergunr100** [comment]
I was walking the dog

Message pops on my phone

"Compass is kil"

"No"

**Comment by Spicierspace153** [comment]
i think there may be a typo there
it should be

"Compass is kil"

"Yes"

**Pull request merged**